### PR TITLE
65 subject encounters

### PR DESCRIPTION
**Pull request by amolnirmalpsl**
ZenHub Id - https://app.zenhub.com/workspaces/avni-data-entry-app-5dee574a1d6415443d9b3deb/issues/avniproject/avni-server/65

### CODE REVIEW

**Comment by sidsamanvay** [comment]
@amolnirmalpsl Could you please squash your commits into one?

**Comment by hithacker** [comment]
Hi @amolnirmalpsl ,
Few comments:

- You are putting encounter data in ProgramEncountersContract which is wrong as Encounter is different from ProgramEncounter. So you should be using EncounterContract.
- In the constructEncounters method in IndividualService you are setting operationalEncounterTypeName from programEncounter.getEncounterType().getName(). `programEncounter.getEncounterType().getName()` is wrong because it returns the name of encounter type and not operational encounter type. Use `programEncounter.getEncounterType().getOperationalEncounterTypeName()` instead to get correct data. Each encounter type will have corresponding operational encounter type.

File: openchs-server-api/src/main/java/org/openchs/service/IndividualService.java [hunk 1]
```diff
@@ -84,9 +96,21 @@ public IndividualContract getSubjectInfo(String individualUuid) {
         }).collect(Collectors.toList());
     }
 
+    public Set<EncounterContract> constructEncounters(Set<Encounter> encounters) {
+        return encounters.stream().map(programEncounter -> {
```
Comment (hithacker) [inline openchs-server-api/src/main/java/org/openchs/service/IndividualService.java]: @amolnirmalpsl variable name is still programEncounter.

File: openchs-server-api/src/main/java/org/openchs/service/IndividualService.java [hunk 2]
```diff
@@ -84,9 +96,21 @@ public IndividualContract getSubjectInfo(String individualUuid) {
...
+            EncounterContract encountersContract = new EncounterContract();
+            encountersContract.setUuid(programEncounter.getUuid());
+            encountersContract.setName(programEncounter.getName());
+            encountersContract.setOperationalEncounterTypeName(programEncounter.getEncounterType().getOperationalEncounterTypeName());
+            encountersContract.setEncounterDateTime(programEncounter.getEncounterDateTime());
+            encountersContract.setEarliestVisitDateTime(programEncounter.getEarliestVisitDateTime());
+            encountersContract.setMaxVisitDateTime(programEncounter.getMaxVisitDateTime());
```
Comment (hithacker) [inline openchs-server-api/src/main/java/org/openchs/service/IndividualService.java]: map the voided field as well. Raise a separate PR to do it also in all the new apis that you have written.

File: openchs-server-api/src/main/java/org/openchs/service/IndividualService.java [hunk 3]
```diff
@@ -32,6 +32,17 @@ public IndividualService(ConceptRepository conceptRepository, IndividualReposito
         this.individualRepository = individualRepository;
     }
 
+    public  IndividualContract getSubjectEncounters(String individualUuid){
+        Individual individual = individualRepository.findByUuid(individualUuid);
+        if (!Objects.nonNull(individual)) {
```
Comment (hithacker) [inline openchs-server-api/src/main/java/org/openchs/service/IndividualService.java]: Is there any reason we are using Objects.nonNull or Objects.isNull? Will not just `individual == null` or `individual != null` suffice?

**Pull request merged**